packages/peregrine/lib/talons/Header/useAccountTrigger.js

```diff
@@ -47,24 +56,82 @@ export const useAccountTrigger = props => {
         // would be to call apolloClient.resetStore() but that would require
         // a large refactor.
         history.go(0);
-    }, [apolloClient, history, revokeToken, setAccountMenuIsOpen, signOut]);
+    }, [
+        apolloClient,
+        history,
+        revokeToken,
+        setAccountMenuIsOpen,
+        signOut,
+        VIEWS.SIGNIN
+    ]);
 
     const handleTriggerClick = useCallback(() => {
         // Toggle the Account Menu.
         setAccountMenuIsOpen(isOpen => !isOpen);
-    }, [setAccountMenuIsOpen]);
+        /**
+         * When the dropdown closes, the view has to
+         * reset to SIGNIN view if the user is not
+         * signed in.
+         */
+        if (!isUserSignedIn) {
+            setView(VIEWS.SIGNIN);
```

Comment:
What if I'm creating an account, get an error, and then accidentally close the dropdown by clicking outside of it? When I go to re-open the form through the account trigger all of my input values will be gone! I think this needs some re-thinking. Since state is persistent we could keep whatever was previously opened as the rendered child and provide navigation buttons similar to how the left drawer nav works with account chip.

Reply:
It was a UX requirement to reset the view when the dropdown is closed. The point you raised it totally acceptable. @schensley any thoughts on this?